core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestValidateQueryAction.java

```diff
@@ -67,58 +68,66 @@ public RestValidateQueryAction(Settings settings, RestController controller, Ind
     }
 
     @Override
-    public void handleRequest(final RestRequest request, final RestChannel channel, final NodeClient client) throws Exception {
+    public Runnable doRequest(final RestRequest request, final RestChannel channel, final NodeClient client) throws Exception {
         ValidateQueryRequest validateQueryRequest = new ValidateQueryRequest(Strings.splitStringByCommaToArray(request.param("index")));
         validateQueryRequest.indicesOptions(IndicesOptions.fromRequest(request, validateQueryRequest.indicesOptions()));
         validateQueryRequest.explain(request.paramAsBoolean("explain", false));
+        validateQueryRequest.types(Strings.splitStringByCommaToArray(request.param("type")));
+        validateQueryRequest.rewrite(request.paramAsBoolean("rewrite", false));
+
         if (RestActions.hasBodyContent(request)) {
             try {
-                validateQueryRequest
-                        .query(RestActions.getQueryContent(RestActions.getRestContent(request), indicesQueriesRegistry, parseFieldMatcher));
-            } catch(ParsingException e) {
-                channel.sendResponse(buildErrorResponse(channel.newBuilder(), e.getDetailedMessage(), validateQueryRequest.explain()));
-                return;
-            } catch(Exception e) {
-                channel.sendResponse(buildErrorResponse(channel.newBuilder(), e.getMessage(), validateQueryRequest.explain()));
-                return;
+                validateQueryRequest.query(
+                    RestActions.getQueryContent(RestActions.getRestContent(request), indicesQueriesRegistry, parseFieldMatcher));
+            } catch (ParsingException e) {
+                return () -> handleException(channel, validateQueryRequest, e.getDetailedMessage());
```

Comment:
I think it should be fine to just throw the exception here. So long as we haven't gone async this **should** be safe. We do it in other places. I guess the only difference is whether we report errors in the query first or extra parameters first which isn't really a big deal either way to me.

This does get me to thinking - maybe we should return a `Consumer<RestChannel>` instead of a `Runanble`? Maybe `doRequest` shouldn't have the variable it needs to send a response so it is never tempted?

Also the name `doRequest`. Maybe `prepare` or something? I'm much better at saying "I don't like that name" than I am at making a better name.